src/mon/Monitor.cc

```diff
@@ -2006,7 +2006,12 @@ void Monitor::handle_probe_reply(MonOpRequestRef op)
 	       << ", mine was " << monmap->get_epoch() << dendl;
       delete newmap;
       monmap->decode(m->monmap_bl);
-      notify_new_monmap(false);
+      if (!has_ever_joined) {
+        dout(20) << " !has_ever_joined " << *monmap << dendl;
+        notify_new_monmap(false, false);
+      } else {
+        notify_new_monmap(false);
+      }
```

Comment:
I'm a little concerned about hitting a similar issue here compared to the above branch.

If we've been dead and missed 3 monmap epochs, the removed_ranks member is just junk to us. I think if we have a map discontinuity, we need to reset our local tracking.

Reply:
Resolved by going into elector.notify_clear_peer_state(); no matter what